SOC-1153 Use a flag to stop this task from sending a welcome message.

garthwebb opened the pull request:
Don't rely on a bunch of heuristics alone.  We send this once and never again.

Also some clean up of surrounding code.

https://wikia-inc.atlassian.net/browse/SOC-1153

Commit:
SOC-1153 Use a flag to stop this task from sending a welcome message.

Don't rely on a bunch of heuristics alone.  We send this once and never again.

Also some clean up of surrounding code.

garthwebb commented on extensions/wikia/HAWelcome/HAWelcomeHooks.php:
```diff
@@ -23,186 +23,6 @@ public static function queueHAWelcomeTask( $wgCityId, $oTitle, $aParams ) {
 	}
 
 	/**
```
The below code was removed because it was superseded by a new hook handler (see https://github.com/Wikia/app/pull/8042/files#diff-7d403cb912d0576b398c56344730ae26L51)

drsnyder replied:
Looks right. This was the previous version before the port to the new job queue.

Commit:
SOC-1153 Whoops, I guess the CityID was already being set to a default of the current wiki

garthwebb commented:
@drsnyder, @nmonterroso, can you take a look at this PR?  I added a `setLocalFlag` method to go with the existing `setLocalFlag` method.  I used this to set a flag to control sending out this HAWelcome message.  I also cleaned up code that was marked as deprecated.

drsnyder commented on extensions/wikia/HAWelcome/HAWelcomeTaskHookDispatcher.php:
```diff
@@ -24,13 +26,32 @@ class HAWelcomeTaskHookDispatcher {
+		if ( $this->currentUser->getLocalFlag( self::WELCOME_SENT_FLAG ) ) {
+			return true;
+		}
+		$this->currentUser->setLocalFlag( self::WELCOME_SENT_FLAG, true );
```
The attribute, preference, flag separation is coming in handy!

drsnyder commented:
LGTM, @nmonterroso?

nmonterroso commented on extensions/wikia/UserLogin/WikiaConfirmEmailSpecialController.class.php:
```diff
@@ -7,7 +7,7 @@
  */
 class WikiaConfirmEmailSpecialController extends WikiaSpecialPageController {
-	const WELCOME_EMAIL_CONTROLLER = 'Email\Controller\Welcome';
+	const WELCOME_EMAIL_CONTROLLER = Email\Controller\WelcomeController::class;
```
Yay class name resolution!

nmonterroso commented:
LGTM

Commit:
SOC-1157 Make sure this class is not abstract

Commit:
SOC-1153 Remove method that checked for 'recently' welcomed, updated tests

We don't care about recently, they were either welcomed or not.  If they
were welcomed, then don't welcome them again.

Commit:
Merge branch 'dev' into SOC-1153
